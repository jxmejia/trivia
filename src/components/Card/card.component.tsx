import { memo } from "react";
import type { CardProps } from "./card.definition";
import { StyledCard } from "./card.style";

const Card = ({ children }: CardProps): JSX.Element => {
  return <StyledCard>{children}</StyledCard>;
};

export default memo(Card);
