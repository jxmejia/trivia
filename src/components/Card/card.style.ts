import styled from "@emotion/styled";
import theme from "../../styles/theme";

export const StyledCard = styled.div`
  border: 1px solid ${theme.colors.black};
  padding: 10px;
  text-align: center;
`;
