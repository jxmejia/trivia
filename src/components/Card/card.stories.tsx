import { ComponentStory, ComponentMeta } from "@storybook/react";
import Card from "./card.component";

export default {
  title: "Card",
  component: Card,
} as ComponentMeta<typeof Card>;

const Template: ComponentStory<typeof Card> = (args) => <Card {...args} />;

export const Default = Template.bind({});

Default.args = {
  children: "Trivia challenge",
};
