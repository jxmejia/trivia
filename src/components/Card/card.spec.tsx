import { createSerializer } from "@emotion/jest";
import { render } from "@testing-library/react";
import Card from "./card.component";

expect.addSnapshotSerializer(createSerializer());

describe("Card", () => {
  it("renders component", () => {
    const { container } = render(<Card />);
    expect(container).toMatchSnapshot();
  });
});
