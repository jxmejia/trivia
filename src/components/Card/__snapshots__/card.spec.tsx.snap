// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card is renders component 1`] = `
.emotion-0 {
  border: 1px solid #000000;
  padding: 10px;
  text-align: center;
}

<div>
  <div
    class="emotion-0"
  />
</div>
`;

exports[`Card renders component 1`] = `
.emotion-0 {
  border: 1px solid #000000;
  padding: 10px;
  text-align: center;
}

<div>
  <div
    class="emotion-0"
  />
</div>
`;
